modules/solid_mechanics/include/neml2/interfaces/NEML2ModelInterface.h

```diff
@@ -74,6 +95,14 @@ NEML2ModelInterface<T>::validParams()
       ":<device-index> optionally specifies a device index. For example, device='cpu' sets the "
       "target compute device to be CPU, and device='cuda:1' sets the target compute device to be "
       "CUDA with device ID 1.");
+
+  params.addParam<std::string>(
```

Comment:
MooseEnum here for possible schedulers?

Reply:
This is the name of the scheduler name defined in the neml2 input file. It is reference an object with user-defined options, so it cannot be an enum.